packages/curvenote-cli/src/submissions/submit.utils.ts

```diff
@@ -510,14 +510,42 @@ export async function getAllSubmissionsUsingKey(
   key: string,
 ): Promise<SubmissionsListItemDTO[] | undefined> {
   session.log.debug(`checking for existing submission using key "${key}"`);
-  let submissions: SubmissionsListingDTO;
+  const submissions: SubmissionsListItemDTO[] = [];
   try {
-    submissions = await getFromJournals(session, `/sites/${venue}/submissions?key=${key}`);
+    const siteSubmissions: SubmissionsListingDTO = await getFromJournals(
+      session,
+      `/sites/${venue}/submissions?key=${key}`,
+    );
+    submissions.push(...siteSubmissions.items);
+  } catch (err) {
+    session.log.debug(err);
+  }
+  try {
+    const mySubmissions: SubmissionsListingDTO = await getFromJournals(
+      session,
+      `/my/submissions?key=${key}`,
+    );
+    // These extra fetches are required for the old version of the API where
+    // the 'key' query parameter is not respected on /my/submissions.
+    // This may be removed (along with the 'correctKey' check below) once
+    // the API is updated.
+    const works = await Promise.all(
+      mySubmissions.items.map((sub) => {
+        return getFromUrl(session, sub.links.work);
```

Comment:
Since `key` is only on `work`, not `submission`, these extra API calls are required to filter the submission list.

On the next release, the API will be updated to correctly respect the `key` query on `/my/submissions` and we can delete these calls.